### PR TITLE
New API init function that accepts config content instead of filename

### PR DESCRIPTION
**Pull request by patriotyk**
@cajun-rat @tjamison 
In aktualizr we use those changes to avoid to include separate config file for librvi, so maybe those changes could be useful for someone another. It is also better for us to use upstream version of library without patch it.

### CODE REVIEW

**Comment by gunnarx** [comment]
+1 on the idea - makes sense to me.  I had a few questions/comments on the code.

File: src/rvi.c [hunk 1]
```diff
@@ -952,9 +952,9 @@ int rviValidateCredential( TRviHandle handle, const char *cred, X509 *cert )
  * Initialize the RVI library. Call before using any other functions.
  */
 
-TRviHandle rviInit ( char *configFilename )
+TRviHandle rviJsonInit ( char *configContent )
```
Comment (gunnarx) [inline src/rvi.c]: Can configContent be const?

File: src/rvi.c [hunk 2]
```diff
@@ -1050,6 +1050,38 @@ TRviHandle rviInit ( char *configFilename )
     return NULL;
 }
 
+
```
Comment (gunnarx) [inline src/rvi.c:1043]: Wondering about this new rviInit implementation.

At first it seems like it would make sense to implement rviInit by first reading a file and then delegating the work to rviJsonInit, just as you have done.   But in terms of the resulting code length I wonder:

The original rviInit was delegating all the file handling to json_load_file()  Isn't it better to continue doing that than to write new file handling code?  

Did you implement some advantage here compared to json_load_file?

File: src/rvi.c [hunk 3]
```diff
@@ -1050,6 +1050,38 @@ TRviHandle rviInit ( char *configFilename )
...
+TRviHandle rviInit (char *configFilename)
```
Comment (gunnarx) [inline src/rvi.c]: The filename ought to be const, right?  (But this is unrelated to the PR I admit -- consider this question more to @tjamison)

Also, technically this is an API change for an API we have actually locked down in specification, so I wonder....    I guess it is a pretty safe change going in that direction though.

**Comment by patriotyk** [comment]
@gunnarx  Please ask Your questions/comments.

**Comment by gunnarx** [comment]
Sorry, did it yesterday but apparently I forgot the "submit" button.

File: src/rvi.c (hunk 1 above)
Comment (patriotyk) [reply src/rvi.c]: Yes, I didn't add const, because for some reasons there wasn't const in rviInit call

File: src/rvi.c (hunk 3 above)
Comment (patriotyk) [reply src/rvi.c]: No, there wasn't const, but I think it should be. And I don't  know the reason why it is not const.

File: src/rvi.c (hunk 2 above)
Comment (patriotyk) [reply src/rvi.c:1043]: I did it because it requiters less changes and no need to write additional internal function. But maybe You are right, I will rewrite it without custom file reading.

Comment (patriotyk) [reply src/rvi.c:1043]: I have updated PR with changes that You suggested.

**Comment by gunnarx** [comment]
Looks good to me now, although the internal helper function should be static.  (I think I found many others when I looked earlier, but that's for a later time)

**Comment by tjamison** [comment]
Looks good to me. It changes the API in a non-breaking way, and since @gunnarx's okay with it, I'll merge.
